test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -0,0 +1,203 @@
+        echo "Release \"$1\" verifying \"$2\""
+        echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+
+        cd "$1"
+        for am in $3; do
+            echo "$1/wt verifying $2 access method $am..."
+            dir="$2/test/format/RUNDIR.$am"
+            WIREDTIGER_CONFIG="$EXT" ./wt -h "../$dir" verify table:wt
+        done
+}
+
+#############################################################
+# upgrade_downgrade:
+#       arg1: branch name #1
+#       arg2: branch name #2
+#       arg3: access methods list
+#############################################################
+upgrade_downgrade()
+{
+        echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+        echo "Upgrade/downgrade testing with \"$1\" and \"$2\""
+        echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+
+        # Alternate running each branch format test program on the second branch's build.
+        # Loop twice, that is, run format twice using each branch.
+        top="$PWD"
+        for am in $3; do
+            for reps in {1..2}; do
+                echo "$1 format running on $2 access method $am..."
+                cd "$top/$1/test/format"
+                flags="-1qR $(format_b_flag $1)"
+                ./t $flags -h "$top/$2/test/format/RUNDIR.$am" timer=2
+
+                echo "$2 format running on $2 access method $am..."
+                cd "$top/$2/test/format"
+                flags="-1qR $(format_b_flag $2)"
+                ./t $flags -h "RUNDIR.$am" timer=2
+            done
+        done
+}
+
+# Create a directory in which to do the work.
+top="test-compatibility-run"
+rm -rf "$top" && mkdir "$top"
+cd "$top"
+
+# Build the branches.
+(build_branch mongodb-3.4)
+(build_branch mongodb-3.6)
+(build_branch mongodb-4.0)
+(build_branch mongodb-4.2)
+(build_branch mongodb-4.4)
+(build_branch develop)
+
+# Get the names of the last two WiredTiger releases, wt1 is the most recent release, wt2 is the
+# release before that. Minor trickiness, we depend on the "develop" directory already existing
+# so we have a source in which to do git commands.
+cd develop; wt1=$(get_prev_version 1); cd ..
+(build_branch "$wt1")
+cd develop; wt2=$(get_prev_version 2); cd ..
+(build_branch "$wt2")
+
+# Run format in each branch for supported access methods.
+(run_format mongodb-3.4 "fix row var")
+(run_format mongodb-3.6 "fix row var")
+(run_format mongodb-4.0 "fix row var")
+(run_format mongodb-4.2 "fix row var")
+(run_format mongodb-4.4 "row")
+(run_format develop "row")
+(run_format "$wt1" "fix row var")
+(run_format "$wt2" "fix row var")
+
+# Verify backward compatibility for supported access methods.
+(verify_branches mongodb-3.6 mongodb-3.4 "fix row var")
+(verify_branches mongodb-4.0 mongodb-3.6 "fix row var")
+(verify_branches mongodb-4.2 mongodb-4.0 "fix row var")
+### (verify_branches mongodb-4.4 mongodb-4.2 "fix row var")
+### (verify_branches develop mongodb-4.4 "row")
```

Comment:
I still not quite understand why `mongodb-4.4` won't work and being commented out. My understanding is if `develop` works `mongodb-4.4` should work too, considering both branches have the same set of wiredtiger source (in a slightly delayed fashion).

Reply:
I was assuming the breakage was because you haven't merged Sue's latest compatibility changes into mongodb-4.4 (develop has never compatibility code). But, I can't justify that logically, and there was a bug in the script I found after I saw those failures.

I'll rerun the test and see if it's OK to turn on the mongodb-4.4 branch.

Reply:
@lukech, the `mongodb-4.4` branch is still not working, `mongodb-4.2` can't verify databases it creates.

I'll chase this and see what I can figure out.